nncf/common/logging/track_progress.py

```diff
@@ -116,13 +117,13 @@ def get_weighted_completed(task: Task, completed: float) -> float:
         """
         if completed % 1 != 0:
             raise Exception(f"Unexpected `completed` value: {completed}.")
-        return sum(task.fields["weights"][: int(completed)])
+        return float(sum(task.fields["weights"][: int(completed)]))
 
 
-class track:
+class track(Generic[ProgressType]):
     def __init__(
         self,
-        sequence: Optional[Union[Sequence[ProgressType], Iterable[ProgressType]]] = None,
+        sequence: Union[Sequence[ProgressType], Iterable[ProgressType], None] = None,
```

Comment:
There is some problem in case of sequence is None, in this case ProgressType is unknown, as result mypy shows var-annotated error.

```python
with track(iterable, description="Processing...") as pbar:
 # error: Need type annotation for "pbar"  [var-annotated]
```

In python3.13 TypeVar have default argument https://peps.python.org/pep-0696/
But now the simplest solution use `with track[None](...)` or add `# type: ignore[var-annotated]`

Reply:
Is `# type: ignore[var-annotated]` would be needed every time `with track(...)` is used? Is there a possibility to ignore this annotation at definition?

Reply:
> Would # type: ignore[var-annotated] be needed every time with track(...) is used?
Yes, comment or [none] 

>  Is there a possibility to ignore this annotation at definition?
It's correctly will works in python3.13 but 3.12 will supported until 2028 😞 
But for python3.9 i dont know how to resolve it